### ISSUE TITLE
build failure on NetBSD (likely due to new in-place edit code)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#16279]
Migrated from [rt.perl.org#132506](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=132506) (status was 'resolved')

Searchable as RT132506$

**Comment by p5pRT** [comment]
### From @tonycoz

#### Created by @tonycoz



While building on NetBSD 6\.0\.1\, I see the following error&#8203;:

\./miniperl \-Ilib make\_ext\.pl lib/auto/DB\_File/DB\_File\.so  MAKE="make" LIBPERL\_A=libperl\.a LINKTYPE=dynamic
Parsing config\.in\.\.\.
Looks Good\.
Can't rename DB\_File\.pm to DB\_File\.pm\.bak&#8203;: Function not implemented\, skipping file at Makefile\.PL line 358\, \<> line 2354\.
Unsuccessful Makefile\.PL\(cpan/DB\_File\)&#8203;: code=19968 at make\_ext\.pl line 518\.
\*\*\* Error code 2

A simple in\-place edit also fails&#8203;:

netbsd601amd64$ \./miniperl \-i\.bak \-pe 's/foo/quux/' bar
Can't rename bar to bar\.bak&#8203;: Function not implemented\, skipping file\, \<> line 1\.

Perl builds successfully on NetBSD 7\.1\.

I'll work on a fix for this\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.27.7:

Configured by tony at Mon Nov 27 20:28:32 EST 2017.

Summary of my perl5 (revision 5 version 27 subversion 7) configuration:
  Commit id: 2cb35ee012cfe486aa75a422e7bb3cb18ff51336
  Platform:
    osname=netbsd
    osvers=6.0.1
    archname=amd64-netbsd
    uname='netbsd netbsd601amd64.tony.develop-help.com 6.0.1 netbsd 6.0.1 (generic) amd64 '
    config_args='-des -Dusedevel'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/pkg/include -D_FORTIFY_SOURCE=2'
    optimize='-O'
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/pkg/include'
    ccversion=''
    gccversion='4.5.3'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -Wl,-rpath,/usr/pkg/lib -Wl,-rpath,/usr/local/lib -fstack-protector -L/usr/pkg/lib'
    libpth=/usr/include/gcc-4.5 /usr/lib /usr/pkg/lib /lib
    libs=-lpthread -lm -lcrypt -lutil -lc -lposix
    perllibs=-lpthread -lm -lcrypt -lutil -lc -lposix
    libc=/lib/libc.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E '
    cccdlflags='-DPIC -fPIC '
    lddlflags='--whole-archive -shared  -L/usr/pkg/lib -fstack-protector'



@INC for perl 5.27.7:
    lib
    /usr/lib/perl5/site_perl/5.27.7/amd64-netbsd
    /usr/lib/perl5/site_perl/5.27.7
    /usr/lib/perl5/5.27.7/amd64-netbsd
    /usr/lib/perl5/5.27.7


Environment for perl 5.27.7:
    HOME=/home/tony
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/tony/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R7/bin:/usr/X11R6/bin:/usr/pkg/bin:/usr/pkg/sbin:/usr/games:/usr/local/bin:/usr/local/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/sh

```
</details>

**Comment by p5pRT** [comment]
### From @tonycoz

On Sun\, 26 Nov 2017 15&#8203;:30&#8203;:34 \-0800\, tonyc wrote&#8203;:
> While building on NetBSD 6\.0\.1\, I see the following error&#8203;:
> 
> \./miniperl \-Ilib make\_ext\.pl lib/auto/DB\_File/DB\_File\.so  MAKE="make"
> LIBPERL\_A=libperl\.a LINKTYPE=dynamic
> Parsing config\.in\.\.\.
> Looks Good\.
> Can't rename DB\_File\.pm to DB\_File\.pm\.bak&#8203;: Function not implemented\,
> skipping file at Makefile\.PL line 358\, \<> line 2354\.
> Unsuccessful Makefile\.PL\(cpan/DB\_File\)&#8203;: code=19968 at make\_ext\.pl line
> 518\.
> \*\*\* Error code 2
> 
> A simple in\-place edit also fails&#8203;:
> 
> netbsd601amd64$ \./miniperl \-i\.bak \-pe 's/foo/quux/' bar
> Can't rename bar to bar\.bak&#8203;: Function not implemented\, skipping file\,
> \<> line 1\.
> 
> Perl builds successfully on NetBSD 7\.1\.
> 
> I'll work on a fix for this\.

This was fixed by 184f90dc410cf3e66a5e682337ae54b3c9f9311f\,  which broke Win32\, which was fixed by a06de4dcf2448ead9a5f5dbd87366727ebd91aca\, 3c67ad9b9e6d659feb76f3acfc9f81cfe59e3660 and 8779d80ae74c415b57b9d139bb415e167292cd2a\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'new' to 'resolved'
